### ISSUE TITLE
Possibly unnecessary (transitive) new Python2 dependency on the qtwebengine_bin-5.15.2 package.

### ISSUE THREAD

**Issue by OscarL** [haikuports/haikuports#6717]
Not sure if this is the right place to mention this, but I've noticed that updating to the latest QTWebEngine_bin package, introduces a dependency on the Python2.7 package (as it is needed by one of QTWebEngine_bin new dependencies, "libevent2").

I found that really weird, why would it need Python2.7 (already have 3.7, 3.8, and 3.6 installed due to some off-tree packages I've tested).

Seems to me that libevent2 needs python2 just for a single Python script. But that script seems to be useful only for developers using libevent2, not for end users of the said library.

Not sure what the correct action would be... if offering that mentioned libevent2 script in a "devel" package, or just patching it to depend on Python3 instead (at quick glance, even running 2to3.py on it might be all what it's needed).

Edit; "event_rpcgen.py" is the name of the libevent2 script that seems to require Python2.x

Notice that the "libevent-1.4" recipe also has that script (and requires Python2.x), but that libevent-2.0.21 recipe requires Python3.7.

**Comment by pulkomandy** [comment]
Probably both: make sure the script runs on python3, and move it to the devel package.

**Comment by OscarL** [comment]
In order to move the `event_rpcgen.py` script from the regular .hpkg to the _devel.hpkg... would it be enough to move the line

    	cmd:event_rpcgen.py

from the `PROVIDES=` section to the `PROVIDES_devel=` one on the recipe? (Sorry, I'm a HaikuPorter neophyte.

Also, I've noticed that the `libevent-1.4.14b.recipe` recipe's REQUIRES section does not lists Python as a dependency.

It does in the case in the `libevent-2.0.21.recipe` one (`cmd:python3.7`), which seems odd, as the said `event_rpcgen.py` script is written in Python2 (print as statement, for example), at least .

As mentioned in my original issue, `libevent21-2.1.8.recipe` lists `cmd:python` as a requirement.

Wouldn't a good first step make sure that all three recipes move `event_rpcgen.py` to the `PROVIDES_devel=`, and the moving the Python dependency to the `REQUIRES_devel=`? I guess that would solve my original issue (dragging in an old Python2 dependency that seems unnecessary).

I guess that this issue should be more appropriately be titled "libevent (1.4, 2.0, 2.1) recipes python dependency issues" at this point, as the association with the `qtwebengine_bin-5.15.2` package was just how I discovered the issue (after installing Falkon).

Everyone of you is already busy, and this is low priority, so I ask these questions to try to understand if it's something even I could provide a PR for. Would it be OK to just make the mentioned changes by editing the recipes via github in that case? One PR per recipe? (I rather leave that to people that know what they are doing much better than me, but... :-D).

Thanks for your time and work, folks.

**Comment by korli** [comment]
The script is now moved to the devel packages.

**Comment by OscarL** [comment]
After @korli's changes, I've updated my beta3 installation (the one where I found the issue after installing Falkon). All looks good there.

I've also installed Falkon (that pulls qtwebengine_bin, that requires libevent2.1) on a fresh nightly installation. No python2 was installed :-)

Great! This is resolved, as far as I'm concerned. Thank you @korli!!!

Should I close the issue? or leave that to the devs?
